au4/src/projectscene/view/timeline/timelineruler.cpp

```diff
@@ -22,13 +22,22 @@ using namespace au::projectscene;
 TimelineRuler::TimelineRuler(QQuickItem* parent)
     : QQuickPaintedItem(parent)
 {
+    setMode(configuration()->timelineRulerMode());
+
     uiconfiguration()->currentThemeChanged().onNotify(this, [this]() { update(); });
-    setFormatter(std::make_unique<BeatsMeasuresFormat>());
+    configuration()->timelineRulerModeChanged().onReceive(this, [this](const TimelineRulerMode mode){
+        setMode(mode);
+        update();
+    });
 }
 
-void TimelineRuler::setFormatter(std::unique_ptr<IRulerFormat> formatter)
+void TimelineRuler::setMode(const TimelineRulerMode mode)
 {
-    m_formatter = std::move(formatter);
+    if (mode == TimelineRulerMode::MINUTES_AND_SECONDS) {
+        m_mode = std::make_unique<TimeFormat>();
```

Comment:
It's better to keep the formatter as it was here. (m_mode -> m_formatter) 
The logic is simple.

There are show modes
There are formatters
For different modes, different formatters
(but ​​in theory there may be for some modes, if more appear, the same formatter, but the difference will be in something else)